core/src/main/java/org/akaza/openclinica/job/XsltTransformJob.java

```diff
@@ -170,24 +170,36 @@ protected void executeInternal(JobExecutionContext context) {
             /**
              *  exportFileName example:EXCEL_ggg_2020-12-14-131854158.xls
              *  Temporally change the value of exportFileName, postProcExportName in epBean with current time stamp 
+             *             
              */
-            if(deleteOld) {
+            
+            String filedescription = epBean.getFiledescription();
+            if(deleteOld ) {
             	 String[] exportFileName = epBean.getExportFileName();
             	 String simpleDatePattern = "yyyy-MM-dd-HHmmssSSS";
             	 SimpleDateFormat sdfDir = new SimpleDateFormat(simpleDatePattern);
             	 String latestTimestr = sdfDir.format(new java.util.Date());
-            	 String fileType = "";
-            	 int i =0;
-            	 while(i < exportFileName.length) {
-            		 fileType = FilenameUtils.getExtension(exportFileName[i]);
-            		 exportFileName[i] = exportFileName[i].substring(0, exportFileName[i].lastIndexOf("_")) + "_" + latestTimestr+ "." + fileType;;
-            		 i++;
-            	 }
-            	 epBean.setExportFileName(exportFileName);
-            	 epBean.setPostProcExportName(exportFileName[0]);
-            	 epBean.setDoNotDelFiles(exportFileName);
+            	 String fileType = "";            	 
             	 
-            	 dataMap.put(POST_FILE_NAME,exportFileName[0]);
+            	 /**
+            	  *  FOR SAS:
+                  *  extract.12.exportname=SAS_MAP.xml,SAS_DATA.xml,SAS_FORMAT.sas
+                  *  no time stamp to replace in those file name
+            	  */
+            	 if(filedescription.indexOf("SAS Data") > -1) {            		 

```

Comment:
Is there a better way to figure out the SAS extract type?

Reply:
I double checked all related information to differentiate SAS dataset, looks like there is no better way to figure out the SAS extract type, because all those information come from extract.poperties file like below:
we can also choose other information which has SAS key word :
extract.12.helpText=Generate SAS Data and Scripts
extract.12.location=$exportFilePath/$datasetName/SAS
extract.12.exportname=SAS_MAP.xml,SAS_DATA.xml,SAS_FORMAT.sas

but it's the same level of:
extract.12.fileDescription=SAS Data and Syntax


============================================

extract.12.odmType=clinical_data
extract.12.file=xml_convert_sas_map.xsl,xml_convert_sas_data.xsl,xml_convert_sas_format.xsl
extract.12.fileDescription=SAS Data and Syntax
extract.12.linkText=Run Now
extract.12.helpText=Generate SAS Data and Scripts
extract.12.location=$exportFilePath/$datasetName/SAS
extract.12.exportname=SAS_MAP.xml,SAS_DATA.xml,SAS_FORMAT.sas
extract.12.zipName=SAS_$datasetName_$dateTime.sas
extract.12.zip=true
extract.12.deleteOld=true
extract.12.success=Your extract job completed successfully. The file is available for download $linkURL.
extract.12.failure=The extract did not complete due to errors. Please contact your system administrator for details.

Reply:
Ok, thanks for checking